File: front/src/app/pages/customer/shared/customer-item.ts
import { Entity } from '../../shared/entity.model';

export class CustomerItem extends Entity {
  constructor(
   public name?: string,
   public address?: string,
   public email?: string,
   public birthDate?: Date
  ) { super(); }

  static fromJson(jsonData: any): CustomerItem {
    return Object.assign(new CustomerItem(), jsonData);
  }
}

File: src/Arch.Cqrs.Client/Models/CustomerModels/CustomerItem.cs
using System;

namespace Arch.Cqrs.Client.Models.CustomerModels
{
    public class CustomerItem
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Email { get; set; }
        public string Address { get; set; }
        public DateTime BirthDate { get; set; }
    }
}